### DISCUSSION TITLE
Cropping: Cropping several parts

### DISCUSSION THREAD

**Discussion by gmourier** [meilisearch/product#350] [Feedback & Feature Proposal]
Today, if a field has to be cropped, the engine stops at the first possible crop. This is not ideal, especially to show the interesting parts of a document from the end user's query.

For example, a document ends with `Currently I'm trying to get nice visuals.` and there is the term `visuals` higher up in this field. 

If the user searches for `get nice visual` in the search bar, this is what the cropper will render in the `_formatted` object:

`loremipsum visual loremipsum`

when it would be more useful to try to display all possible cropped parts for visual comfort/reading information and relevance:

`...loremipsum visual loremipsum... ...nice visuals.`

Depending on the `cropLength` value, several crops could automatically be merged into one if they collide.

We could also propose a mode that is based on the density of the terms found to keep the best crop.

**Comment by curquiza** [comment]
We could also propose a mode that is based on the density of the terms found to keep the best crop, see this issue: https://github.com/meilisearch/meilisearch/issues/1912

**Comment by gmourier** [comment]
The next iteration of cropping favors a single cropped part but is selected according to the density against the query terms found to make it more helpful for readers by choosing the most relevant part.

https://github.com/meilisearch/specifications/pull/120

**Comment by gmourier** [comment]
Following the [v0.28 release](https://github.com/meilisearch/meilisearch/releases/tag/v0.28.0), this discussion is now locked.

We encourage you to create a new thread if needed.

Thank you ✨
